app/controllers/spaces.rb

```diff
@@ -0,0 +1,90 @@
+class MakersBnb < Sinatra::Base
+  get '/spaces' do
+    @date_from = session[:date_from]
+    @date_to = session[:date_to]
+    @today = (Date.today).to_s
+    @tomorrow = (Date.today + 1).to_s
+
+    if @date_from && @date_to
+      @spaces = Space.all(:date_from.lte => @date_from,
+                          :date_to.gte => @date_to)
+    else
+      reset_date_filter
+      @spaces = Space.all
+    end
+    @user = User.get(session[:user])
+    erb :'spaces/spaces'
+  end
+
+  post '/spaces/new' do
+    validate_space_availability(params[:date_from], params[:date_to], '/spaces/new')
+    attributes = {
+      name: params[:name],
+      description: params[:description],
+      price: params[:price],
+      date_from: params[:date_from],
+      date_to: params[:date_to],
+      user_id: current_user.id,
+      image_url: (Cloudinary::Uploader.upload(params['image'][:tempfile],
+                                             crop: :limit, width: 750, height: 500).fetch('url') if params['image'])
+    }
+
+    Space.create(attributes)
+    redirect('/spaces')
+  end
```

Comment:
Ooh, refactor opportunity (I noticed that there were two very similar, non-DRY lines, which gave me the instinct to refactor):

```ruby
post '/spaces/new' do
  space = Space.build(params)
  return redirect('/spaces'), notice: "#{ space.errors ? space.errors : "Space created!" }"
end
```